opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -250,6 +250,12 @@ namespace Opm
             if (use_gpu) {
                 const std::string gpu_mode = EWOMS_GET_PARAM(TypeTag, std::string, GpuMode);
                 WellContributions wellContribs(gpu_mode);
+#if HAVE_OPENCL
+                if(gpu_mode.compare("opencl") == 0){
+                    auto openclBackend = static_cast<bda::openclSolverBackend<block_size>*>(bdaBridge->backend.get());
+                    wellContribs.setOpenCLEnv(openclBackend->context.get(), openclBackend->queue.get());
+                }
+#endif
```

Comment:
Would this kind of setting not be possible directly in `openclSolverBackend, e.g.  `openclSolverBackend::solve_system`? Seems like it has access to `wellContribs`. Or is that too late?

Reply:
It would be too late because in this new implementation the well data gets written to the GPU as it becomes available and this happens before `solve_system` is called.

Reply:
I see